packages/realm-web/src/App.ts

```diff
@@ -0,0 +1,239 @@
+                throw new Error(
+                    `Failed to switch user (id = ${nextUser}) - did you log in?`
+                );
+            }
+        } else if (nextUser instanceof User) {
+            const index = this.users.findIndex(({ user }) => user === nextUser);
+            if (index >= 0) {
+                // Remove the user from the stack
+                const [handle] = this.users.splice(index, 1);
+                // Insert the user in the beginning of the stack
+                this.users.splice(0, 0, handle);
+            } else {
+                throw new Error(
+                    "Cannot switch to a user that was never logged in"
+                );
+            }
+        } else {
+            throw new Error("Expected a user id or a User instance");
+        }
+    }
+
+    /**
+     * Log in a user
+     * @param credentials Credentials to use when logging in
+     * @param fetchProfile Should the users profile be fetched? (default: true)
+     */
+    public async logIn(credentials: Realm.Credentials, fetchProfile = true) {
+        // See https://github.com/mongodb/stitch-js-sdk/blob/310f0bd5af80f818cdfbc3caf1ae29ffa8e9c7cf/packages/core/sdk/src/auth/internal/CoreStitchAuth.ts#L746-L780
+        const response = await this.appTransport.fetch(
+            {
+                method: "POST",
+                path: `/auth/providers/${credentials.providerName}/login`,
+                body: credentials.material
+            },
+            null
+        );
+        // Spread out values from the response and ensure they're valid
+        const {
+            user_id: userId,
+            access_token: accessToken,
+            refresh_token: refreshToken
+        } = response;
+        if (typeof userId !== "string") {
+            throw new Error("Expected a user id in the response");
+        }
+        if (typeof accessToken !== "string") {
+            throw new Error("Expected an access token in the response");
+        }
+        if (typeof refreshToken !== "string") {
+            throw new Error("Expected an refresh token in the response");
+        }
+        // Create the user
+        const handle = User.create({
+            app: this,
+            id: userId,
+            accessToken,
+            refreshToken
+        });
+        // If neeeded, fetch and set the profile on the user
+        if (fetchProfile) {
+            const profile = await this.baseTransport.fetch(
+                { method: "GET", path: "/auth/profile" },
+                handle.user
+            );
+            handle.controller.setProfile(profile);
+        }
+        // Add the user at the top of the stack
+        this.users.splice(0, 0, handle);
+        // Return the user
+        return handle.user;
+    }
+
+    /**
+     * Log out a user
+     * @param userOrId The user or id of the user to log out (default: currentUser)
+     */
+    public async logOut(
+        userOrId: Realm.User | string | null = this.currentUser
```

Comment:
Pasting a comment from above ☝️:

> This will change once we merge #2794 and run the --fix. I simply forgot to add printWidth: 120 to the prettierrc.js added with #2811.

I'm a little tempted to run it on this PR, as a personal favor 😄 Just so you don't have to poke your eyes out from reading the code 😝

Reply:
👍 😅